.github/workflows/lint.yml

```diff
@@ -18,15 +18,15 @@ jobs:
       - uses: actions/setup-go@v3
         with:
           go-version: 1.19
-      - run: ./hack/make engine:lint
+      - run: ./hack/dev ./hack/make engine:lint
```

Comment:
Revisiting this now, one last tweak I think might make sense is something like this:
* `./hack/make engine:test` - results in the dev engine being bootstrapped and running the provided target
   * basically, `./hack/make` always implicitly wraps itself with `./hack/dev` (or the end effect of doing that)
* `./hack/make-prod engine:test` results in the tests running against the actual prod release, basically just skip the dev bootstrap

We can still retain `./hack/dev` too to support wrapping arbitrary commands.

Reply:
Ran into a bunch of problems recently (catch 22): https://github.com/dagger/dagger/pull/3928#discussion_r1027021899

Brainstorming here, but would this be simpler:

- We write our pipelines targeting stable *entirely*
  - separate `go.mod`, there's no multiple version of dagger in the same codebase competing against each other
- In our pipelines, we bootstrap dagger and run tests with dagger-in-dagger-main
  - `internal/mage` -> stable
  - `core/integration`, `sdk/*/*test` -> main

Reply:
While we're talking about this, I'd also _really_ like the ability to run the tests against a native local engine. Having to rebuild the image all the time for engine changes (even just `log.Println`) is a pretty gnarly hit to the feedback loop. :confused:

I'm still for testing against the engine image since that's what actually ships now, and maybe we do that in CI where the feedback loop doesn't matter, but at least for local dev it'd be awesome to optimize.